### PR TITLE
Various minor tweaks and improvements. Needs review.

### PR DESCRIPTION
**Pull request by mfxmfx**
Hi Chris,

this is NOT meant for merging, but maybe you could have a look and pick the pieces
that look good to you.

Thanks,
Markus

### CODE REVIEW

File: vi_motion.lua [hunk 1]
```diff
@@ -100,6 +100,19 @@ local motions = {
   G = { MOV_LINE, vi_motions.goto_line, -1},
   ["'"] = wrap_table(registers, restore_mark),
   ['%'] = { MOV_INC, vi_motions.match_brace, 1 },
+  -- these are included here so that they work with a repeat count:
```
Comment (jugglerchris) [inline vi_motion.lua]: Thanks!  I've taken this change.

**Comment by jugglerchris** [comment]
Ok, I (finally) went through the changes manually merging the ones that looked good.  Turns out they're all good, thanks, so I'll just merge it as is.  Sorry it's taken so long!

**Comment by mfxmfx** [comment]
BTW, do we still need your modified textredux version? It would be great if your changes could be merged upstream, or otherwise please update the docs. Thanks!

**Comment by jugglerchris** [comment]
Yes, the modified textredux is unfortunately still needed, unless I've missed any important changes going in.  I think @rgieseke wanted to do it differently to how I did it (which was easiest for me to do to get it working, rather than necessarily the best overall solution), but maybe he can comment.  I'd be more than happy to ditch my fork and use the proper textredux once it supports the command entry.  :-)

**Comment by mfxmfx** [comment]
Thanks for finally merging textredux - this should make it easier to provide feedback & patches.

**Pull request merged**